annotation/lib/http.dart

```diff
@@ -9,94 +9,150 @@ class HttpMethod {
 }
 
 @immutable
 class Method {
+  /// HTTP request method which can be found in [HttpMethod].
   final String method;
+
+  /// A relative or absolute path, or full URL of the endpoint.
+  ///
+  /// See [RestApi.baseUrl] for details of how this is resolved against a base URL
+  /// to create the full endpoint URL.
   final String path;
   const Method(this.method, this.path);
 }
 
+/// Make a `GET` request
+///
+/// ```
+/// @GET("ip")
+/// Future<Response<String>> ip(@Query('query1') String query)
+/// ```
 @immutable
 class GET extends Method {
   const GET(String path) : super(HttpMethod.GET, path);
 }
 
+/// Make a `POST` request
 @immutable
 class POST extends Method {
   const POST(String path) : super(HttpMethod.POST, path);
 }
 
+/// Make a `PATCH` request
 @immutable
 class PATCH extends Method {
   const PATCH(final String path) : super(HttpMethod.PATCH, path);
 }
 
+/// Make a `PUT` request
 @immutable
 class PUT extends Method {
   const PUT(final String path) : super(HttpMethod.PUT, path);
 }
 
+/// Make a `DELETE` request
 @immutable
 class DELETE extends Method {
   const DELETE(final String path) : super(HttpMethod.DELETE, path);
 }
 
+/// Adds headers specified in the [Map].
 @immutable
 class Headers {
   final Map<String, dynamic> value;
   const Headers([this.value]);
 }
 
+/// Replaces the header with the value of its target.
+///
+/// Header parameters may be `null` which will omit them from the request.
 @immutable
 class Header {
   final String value;
   const Header(this.value);
 }
 
-// @immutable
-// class HeaderMap {
-//   const HeaderMap();
-// }
-
+/// Use this annotation on a service method param when you want to directly control the request body
+/// of a POST/PUT request (instead of sending in as request parameters or form-style request
+/// body).
+///
+/// Body parameters may not be `null`.
 @immutable
 class Body {
   const Body();
 }
 
+/// Named pair for a form request.
+///
+/// ```
+/// @POST("/post")
+/// Future<Response<String>> example(
+///   @Field() int foo,
+///   @Field("bar") String barbar},
+/// )
+/// ```
+/// Calling with `foo.example("Bob Smith", "President")` yields a request body of
+/// `foo=Bob+Smith&bar=President`.
 @immutable
 class Field {
   final String value;
   const Field([this.value]);
 }
 
+/// Named replacement in a URL path segment.
+///
+/// Path parameters may not be `null`.
 @immutable
 class Path {
   final String value;
   const Path([this.value]);
 }
 
+/// Query parameter appended to the URL.
+///
+/// Simple Example:
+///
+///```
+/// @GET("/get")
+/// Future<Response<String>> foo(@Query('bar') String query)
+///```
+/// Calling with `foo.friends(1)` yields `/get?bar=1`.
 @immutable
 class Query {
   final String value;
   final bool encoded;
   const Query(this.value, {this.encoded = false});
 }
 
+/// Query parameter keys and values appended to the URL.
+///
+/// A `null` value for the map, as a key, or as a value is not allowed.
 @immutable
 class QueryMap {
```

Comment:
Because we have `Headers` for multiple `Header`. How about class name `Queries` for multiple `Query`? just for consistency. What do you think?
```suggestion
 class Queries {
```

Reply:
Yeah. It should be better. I will submit this in another commit.